workshops/util.py

```diff
@@ -946,3 +947,20 @@ def homework2state(homework):
     else:  # failed homework
         assert homework.state == 'f'
         return 'none'
+
+
+def redirect_with_next_support(request, *args, **kwargs):
+    """Works in the same way as `redirect` except when there is GET parameter
+    named "next". In that case, user is redirected to the URL from that
+    parameter. If you have a class-based view, use RedirectSupportMixin that
+    does the same. """
+
+    next_url = request.GET.get('next', None)
+    if next_url is not None and is_safe_url(next_url):
+        return redirect(next_url)
+    else:
+        return redirect(*args, **kwargs)
+
+
+def dict_without_Nones(**keys):
+    return {k: v for k, v in keys.items() if v is not None}
```

Comment:
Please check if we don't have something similar used somewhere, because I do seem to recall.